src/nodes.coffee

```diff
@@ -1104,47 +1115,77 @@ exports.Obj = class Obj extends Base
 
   children: ['properties']
 
+  isAssignable: ->
+    @lhs = true
```

Comment:
So I tackled this and ended up with 05c1f46a9237d6b0eb83c0cd8917f2baa8603ded. I traced how `lhs` was getting set in your version (i.e., where was the `isAssignable` call coming from) and just set the property from the parent at that point. The end result came in rather complicated, though, because of the cases of objects within arrays, and of destructuring that originates from function parameters and not `Assign`. I think the performance should be the same as your version, at least.

I’m sure you could find an even more elegant way. I would at least prefer this over an `isAssignable` method that mutates; but feel free to try to do better still 😄